lib/testing-library-detection.ts

```diff
@@ -0,0 +1,52 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+/**
+ * Enhances a given rule with helpers to detect Testing Library utils.
+ */
+function testingLibraryDetection(rule: any) {
+  return (context: any, options: any) => {
+    let isImportingTestingLibrary = false;
+
+    // TODO: init here options based on shared ESLint config
+
+    // helpers for Testing Library detection
+    const helpers = {
+      getIsImportingTestingLibrary() {
+        return isImportingTestingLibrary;
+      },
+    };
+
+    // instructions for Testing Library detection
+    const detectionInstructions = {
+      ImportDeclaration(node: TSESTree.ImportDeclaration) {
+        isImportingTestingLibrary = /testing-library/g.test(
+          node.source.value as string
+        );
+      },
+    };
+
+    // update given rule to inject Testing Library detection
+    const ruleInstructions = rule(context, options, helpers);
```

Comment:
would `options` be where the eslint configuration (for instance: the custom import) is?

Reply:
No, these options are always the rule options. If you want to access shared ones, you'll need to get them through `context`. The idea is not having to check shared options manually within rules, as the TL detection will check which  are those options to decide if a method is from Testing Library or not.